src/InfoBarWidget.js

```diff
@@ -0,0 +1,122 @@
+	OO.ui.mixin.IconElement.call( this, config );
+	OO.ui.mixin.LabelElement.call( this, config );
+	OO.ui.mixin.TitledElement.call( this, config );
+	OO.ui.mixin.FlaggedElement.call( this, config );
+
+	this.closeIcon = new OO.ui.IconWidget( {
+		icon: 'clear',
+		flags: [ 'invert' ],
+		classes: [ 'ext-wwt-infoBarWidget-close' ]
+	} );
+	this.userInfoLabel = new OO.ui.LabelWidget( {
+		label: mw.msg( 'ext-whowrotethat-ready-general' ),
+		classes: [ 'ext-wwt-infoBarWidget-info' ]
+	} );
+	this.$pendingAnimation = $( '<div>' )
+		.addClass( 'ext-wwt-infoBarWidget-spinner' )
+		.append(
+			$( '<div>' )
+				.addClass( 'ext-wwt-infoBarWidget-spinner-bounce' )
+		);
+
+	// Set properties
+	this.setState( config.state || 'pending' );
+	this.setLabel( mw.msg( 'ext-whowrotethat-state-pending' ) );
+
+	// Scroll event
+	$( window ).scroll( () => {
+		var pos = $( '#bodyContent' )[ 0 ].getBoundingClientRect();
+
+		widget.$element.toggleClass(
+			'ext-wwt-infoBarWidget-sticky',
+			pos.top < widget.$element.height()
+		);
+	} );
+
+	// Close event
+	this.closeIcon.$element.on( 'click', function () {
+		widget.emit( 'close' );
+	} );
+
+	// Initialize
+	this.$element
+		.addClass( 'ext-wwt-infoBarWidget' )
+		.append(
+			this.$pendingAnimation,
+			this.$icon,
+			this.$label,
+			this.userInfoLabel.$element,
+			this.closeIcon.$element
+		);
+};
+
+/* Setup */
+
+OO.inheritClass( InfoBarWidget, OO.ui.Widget );
+OO.mixinClass( InfoBarWidget, OO.ui.mixin.IconElement );
+OO.mixinClass( InfoBarWidget, OO.ui.mixin.LabelElement );
+OO.mixinClass( InfoBarWidget, OO.ui.mixin.TitledElement );
+OO.mixinClass( InfoBarWidget, OO.ui.mixin.FlaggedElement );
+
+/**
+ * Define legal states for the widget
+ *
+ * @type {Array}
+ */
+InfoBarWidget.static.legalFlags = [ 'pending', 'ready', 'err' ];
+
+/**
+ * Change the state of the widget
+ *
+ * @param {string} state Widget state; 'pending', 'ready' or 'error'
+ */
+InfoBarWidget.prototype.setState = function ( state ) {
+	var flags = {};
+
+	if ( this.state !== state ) {
+		this.constructor.static.legalFlags.forEach( ( flag ) => {
+			flags[ flag ] = flag === state;
+		} );
+		flags.invert = true;
+
+		this.setFlags( flags );
+
+		if ( state === 'ready' ) {
+			this.setLabel( mw.msg( 'ext-whowrotethat-ready-title' ) );
+			this.userInfoLabel.setLabel( mw.msg( 'ext-whowrotethat-ready-general' ) );
+			this.setIcon( 'userAvatar' );
+		} else if ( state === 'pending' ) {
+			this.setIcon( '' );
+		} else {
+			this.setIcon( 'error' );
+			this.setErrorMessage();
+		}
+
+		this.$pendingAnimation.toggle( state === 'pending' );
+		this.userInfoLabel.toggle( state === 'ready' );
+		this.closeIcon.toggle( state !== 'pending' );
+
+		this.state = state;
+	}
+};
+
+/**
+ * Set an error with a specific label
+ *
+ * @param {string} message Error label message key
+ */
+InfoBarWidget.prototype.setErrorMessage = function ( message ) {
+	message = message || 'ext-whowrotethat-state-error-generic';
```

Comment:
you should be able to set defaults in the argument list, `function ( message = 'ext-whowrotethat-state-error-generic' )`

Reply:
Good point, I forgot I can do that